tool/plugin/lib/plugin.dart

```diff
@@ -632,7 +672,7 @@ class BuildCommand extends ProductCommand {
         processedFile.writeAsStringSync(source);
       }
 
-      if (spec.channel == 'stable') {
+      if (!spec.version.startsWith('4.')) {
```

Comment:
Why check against a specific number? If we need to check against `4.` perhaps document in-line for other readers.

Reply:
That's standard practice; checking stable was a mistake. The edits need to be made regardless of which channel we are building. 

There is a TODO upstream. Someday we should refactor the editing code into a separate object.